src/output.py

```diff
@@ -198,7 +198,17 @@ def appendToFile(command):
 
 
 def appendExit():
-    appendToFile('exit;')
+    # The `$SHELL` environment variable points to the default shell,
+    # not the current shell. But they are often the same. And there
+    # is no other simple and reliable way to detect the current shell.
+    shell = os.environ['SHELL']
+    # ``csh``, fish`` and, ``rc`` uses ``$status`` instead of ``$?``.
+    if shell.endswith('csh') or shell.endswith('fish') or shell.endswith['rc']:
+        exist_status = '$status'
```

Comment:
> never knew about the difference between shell being the default versus being the current one.

The default shell is the current shell if current shell set `$SHELL` when it starts, otherwise the default shell is login shell.

> do you mean exit_status here

Yes. Previous versions uses `exit $?` and it later changed to `exit;` (for compatibility with csh and fish?).